templates/mongodb.conf.erb

```diff
@@ -1,35 +1,36 @@
 #mongodb.conf - generated from Puppet
 
-#System Log
-
+<% config_hash = {} -%>
+<% if @syslog || @logpath || @logappend || @system_logrotate || !@verbose.nil? || @verbositylevel -%>
+<% config_hash['systemLog'] = {} -%>
 <% if @syslog -%>
-systemLog.destination: syslog
+<% config_hash['systemLog']['destination'] = 'syslog' -%>
 <% elsif @logpath -%>
-systemLog.path: <%= @logpath %>
-systemLog.destination: file
+<% config_hash['systemLog']['destination'] = 'file' -%>
+<% config_hash['systemLog']['path'] = @logpath -%>
 <% if @logappend -%>
-systemLog.logAppend: <%= @logappend %>
+<% config_hash['systemLog']['logAppend'] = @logappend -%>
 <% end -%>
 <% if @system_logrotate -%>
-systemLog.logRotate: <%= @system_logrotate %>
+<% config_hash['systemLog']['logRotate'] = @system_logrotate -%>
```

Comment:
The naming of existing params is kind of annoying.
Here the param name all of a sudden is prepended with 'system_' ...